x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/services/AbstractInferenceServiceParameterizedTests.java

```diff
@@ -0,0 +1,387 @@
+
+    public AbstractInferenceServiceParameterizedTests(
+        AbstractInferenceServiceBaseTests.TestConfiguration testConfiguration,
+        TestCase testCase
+    ) {
+        super(testConfiguration);
+        this.testCase = testCase;
+    }
+
+    @Override
+    public InferenceService createInferenceService() {
+        return testConfiguration.commonConfig().createService(threadPool, clientManager);
+    }
+
+    public record TestCase(
+        String description,
+        Function<AbstractInferenceServiceBaseTests.TestConfiguration, Utils.PersistedConfig> createPersistedConfig,
+        ServiceParser serviceParser,
+        TaskType expectedTaskType,
+        boolean modelIncludesSecrets,
+        boolean expectFailure
+    ) {}
+
+    private record ServiceParserParams(
+        SenderService service,
+        Utils.PersistedConfig persistedConfig,
+        AbstractInferenceServiceBaseTests.TestConfiguration testConfiguration
+    ) {}
+
+    @FunctionalInterface
+    private interface ServiceParser {
+        Model parseConfigs(ServiceParserParams params);
+    }
+
+    private static class TestCaseBuilder {
+        private final String description;
+        private final Function<AbstractInferenceServiceBaseTests.TestConfiguration, Utils.PersistedConfig> createPersistedConfig;
+        private final ServiceParser serviceParser;
+        private final TaskType expectedTaskType;
+        private boolean modelIncludesSecrets;
+        private boolean expectFailure;
+
+        TestCaseBuilder(
+            String description,
+            Function<AbstractInferenceServiceBaseTests.TestConfiguration, Utils.PersistedConfig> createPersistedConfig,
+            ServiceParser serviceParser,
+            TaskType expectedTaskType
+        ) {
+            this.description = description;
+            this.createPersistedConfig = createPersistedConfig;
+            this.serviceParser = serviceParser;
+            this.expectedTaskType = expectedTaskType;
+        }
+
+        public TestCaseBuilder withSecrets() {
+            this.modelIncludesSecrets = true;
+            return this;
+        }
+
+        public TestCaseBuilder expectFailure() {
+            this.expectFailure = true;
+            return this;
+        }
+
+        public TestCase build() {
+            return new TestCase(description, createPersistedConfig, serviceParser, expectedTaskType, modelIncludesSecrets, expectFailure);
+        }
+    }
+
+    @ParametersFactory
+    public static Iterable<TestCase[]> parameters() throws IOException {
+        return Arrays.asList(
+            new TestCase[][] {
+                // Test cases for parsePersistedConfig method
+                {
+                    new TestCaseBuilder(
+                        "Test parsing persisted config without chunking settings",
+                        testConfiguration -> getPersistedConfigMap(
```

Comment:
The test configurations have to be passed in because the `parameters()` method must be static